src/test/setupE2EProcaptcha.ts

```diff
@@ -0,0 +1,136 @@
+import {ContractDeployer, getPairAsync, ProsopoCaptchaContract, wrapQuery} from "@prosopo/contract";
+import {ContractAbi, ContractFile, DappPayee, Hash, Payee, RandomProvider} from "@prosopo/captcha-contract";
+import {hexToU8a, stringToU8a} from "@polkadot/util";
+import {Abi} from "@polkadot/api-contract";
+import {randomAsHex} from "@polkadot/util-crypto";
+import {EventRecord} from "@polkadot/types/interfaces";
+import {KeyringPair} from "@polkadot/keyring/types";
+import {ApiPromise} from "@polkadot/api";
+
+export type ProcaptchaTestSetup = { contract: ProsopoCaptchaContract, contractAddress: string, testAccount: string, siteKey: string }
+
+export async function setupProcaptcha(api: ApiPromise, siteKey: string, port:number): Promise<ProcaptchaTestSetup> {
+    try {
+        await api.isReady;
+        const alicePair = await getPairAsync(undefined, '//Alice', undefined, 'sr25519', 42)
+        const contract = await deployProcaptchaContract(api, alicePair);
+        console.log("Captcha contract address", contract.address.toString())
+
+        // Set the calling pair to Bob so that he is registered as the provider (the calling account is registered in the contract as the provider)
+        contract.pair = await getPairAsync(undefined, '//Bob', undefined, 'sr25519', 42);
+
+        await procaptchaProviderRegister(contract, port);
+        console.log(`Registered Bob ${contract.pair.address} as provider`)
+        await procaptchaProviderSetDataset(contract);
+
+        // any account can register an app site key so using Bob will be fine
+        await procaptchaAppRegister(contract, siteKey);
+
+        // get random active provider and block number from response
+        //const randomProvider = await procaptchaGetRandomProvider(contract, siteKey, alicePair.address)
+
+        return {contract, contractAddress: contract.address, testAccount: alicePair.address, siteKey: siteKey}
+    } catch (e) {
+        console.error(e)
+        throw new Error (`Failed to setup Procaptcha: ${JSON.stringify(e, null ,4)}`)
```

Comment:
Rule of thumb: never handle (including logging) and throw errors at the same time. Either one, or another. Otherwise, it leads to double-logging, or to invalid logging (e.g. the error is logged, although it was handled on the upper level)

UPD: do we even need this try-catch and the wrapping at all?
UPD 2: the same point in the methods below

Reply:
Sure, I'll remove the `console.error`.

In terms of the `try` `catch` I thought it added a bit of context to the issue that occurred during setup. Quite happy to remove them and let a higher up `try` `catch` do the work with a more generalised message.

Reply:
I think it's better to remove it. 
`setupProcaptcha` in the stacktrace tells pretty much the same thing about the context.

Reply:
Removed.